ui/src/containers/DisableAppModalContainer.js

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react';
+import { Button, TextInput, Form, FormGroup, Stack, StackItem, Modal } from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { toggleDisableAppModal } from '../actions/actions-ui';
+
+const DisableAppModalContainer = ({ isOpen, toggleDisableAppModal }) => {
+  function handleDisableAppSave () {
+    // TODO: This needs to send a request to the backend and update the client state
+    // this.props.toggleDisableAppModal();
+  };
+
+  const [disableMessage, setDisableMessage] = useState('');
+
+  return (
+    <Modal
+      isLarge
+      title="Disable All App Versions"
+      isOpen={isOpen}
+      onClose={toggleDisableAppModal}
+      actions={[
+        <Button key="cancel" variant="secondary" onClick={toggleDisableAppModal}>
+          Cancel
+        </Button>,
+        <Button key="save" variant="primary" onClick={handleDisableAppSave}>
+          Save
+        </Button>
+      ]}
+    >
+      <Stack gutter="md">
+        <StackItem>
+          You have requested App Disablement. This will disable all current versions of the App. Are you sure you want to proceed?
+        </StackItem>
+        <StackItem isMain>
+          <Form isHorizontal>
+            <FormGroup
+              label="Disabled Message"
+              fieldId="horizontal-form-disable-message"
+              helperText="Optional Field. This will overwrite existing custom disabled messages for all versions"
+            >
+              <TextInput
+                value={disableMessage}
+                onChange={(value) => setDisableMessage(value)}
```

Comment:
Suggest handler be extracted to its own method for performance reasons when component re-renders

Reply:
Interested to see what the performance implications of this would be, if you have any materials on it.